File: tests/runtimes/deno/deno_globs.py
from typegraph.graph.typegraph import Graph
from typegraph.policy import Policy
from typegraph.runtimes.deno import DenoRuntime

from typegraph import t, typegraph


@typegraph()
def deno_globs(g: Graph):
    deno = DenoRuntime()
    public = Policy.public()

    g.expose(
        public,
        test_glob=deno.import_(
            t.struct({"a": t.float(), "b": t.float()}),
            t.float(),
            module="ts/dep/main.ts",
            deps=["ts/**/nested/*.ts"],
            name="doAddition",
        ),
    )

File: tests/runtimes/deno/deno_globs.ts
// Copyright Metatype OÜ, licensed under the Mozilla Public License Version 2.0.
// SPDX-License-Identifier: MPL-2.0

import { Policy, t, typegraph } from "@typegraph/sdk";
import { DenoRuntime } from "@typegraph/sdk/runtimes/deno";

export const tg = await typegraph(
  {
    name: "deno_globs",
  },
  (g) => {
    const deno = new DenoRuntime();
    const pub = Policy.public();

    g.expose({
      testGlob: deno
        .import(t.struct({ a: t.float(), b: t.float() }), t.float(), {
          module: "ts/dep/main.ts",
          name: "doAddition",
          deps: ["ts/dep/**/*.ts"],
        })
        .withPolicy(pub),
    });
  },
);
